### PR TITLE
Support _HOST placeholder in kerberos principal in Hive connector

### PR DESCRIPTION
**Pull request by arhimondr**
Product test has been added.

@supersedes https://github.com/prestodb/presto/pull/6861

Must be merged after https://github.com/Teradata/docker-images/pull/9 is merged and released, and the version is changed from `latest` to `12` in `compose-commons.sh`

### CODE REVIEW

**Comment by amrutagokhale** [comment]
Can you hold on before merging these commits? My changes (for migrating to latest docker image) are in a  state of flux. I added some comments in that file later to explain the changes I made (https://github.com/Teradata/presto/pull/458/commits), but it isn't final yet

File: presto-product-tests/conf/presto/etc/environment-specific-catalogs/singlenode-kerberos-hdfs-impersonation/hive.properties [hunk 1]
```diff
@@ -12,13 +12,13 @@ hive.metastore-cache-ttl=0s
...
-hive.metastore.client.principal=hive/hadoop-master@LABS.TERADATA.COM
-hive.metastore.client.keytab=/etc/hive/conf/hive.keytab
+hive.metastore.client.principal=hive/_HOST@LABS.TERADATA.COM
+hive.metastore.client.keytab=/etc/presto/conf/presto-hive.keytab
 
 hive.hdfs.authentication.type=KERBEROS
 hive.hdfs.impersonation.enabled=true
-hive.hdfs.presto.principal=hdfs/hadoop-master@LABS.TERADATA.COM
-hive.hdfs.presto.keytab=/etc/hadoop/conf/hdfs.keytab
+hive.hdfs.presto.principal=presto-server/_HOST@LABS.TERADATA.COM
```
Comment (cawallin) [inline presto-product-tests/conf/presto/etc/environment-specific-catalogs/singlenode-kerberos-hdfs-impersonation/hive.properties]: please use presto-client/_HOST, we shouldn't use the Presto service principal as the principal to connect to hdfs

File: presto-product-tests/conf/presto/etc/environment-specific-catalogs/singlenode-kerberos-hdfs-impersonation/hive.properties [hunk 2]
```diff
@@ -12,13 +12,13 @@ hive.metastore-cache-ttl=0s
 
 hive.metastore.authentication.type=KERBEROS
 hive.metastore.service.principal=hive/hadoop-master@LABS.TERADATA.COM
-hive.metastore.client.principal=hive/hadoop-master@LABS.TERADATA.COM
-hive.metastore.client.keytab=/etc/hive/conf/hive.keytab
+hive.metastore.client.principal=hive/_HOST@LABS.TERADATA.COM
```
Comment (cawallin) [inline presto-product-tests/conf/presto/etc/environment-specific-catalogs/singlenode-kerberos-hdfs-impersonation/hive.properties]: please use presto-hive-client/_HOSTS or something like that, that makes it clear that the client principal is distinct from the hive service principal

File: presto-product-tests/conf/presto/etc/environment-specific-catalogs/singlenode-kerberos-hdfs-impersonation/hive.properties (hunk 1 above)
Comment (arhimondr) [inline presto-product-tests/conf/presto/etc/environment-specific-catalogs/singlenode-kerberos-hdfs-impersonation/hive.properties]: We use `presto-client` principal to connect to Presto from CLI. So, here `presto-server` principal must be used. By analogy how it is done in Hive or HDFS.

File: presto-product-tests/conf/presto/etc/environment-specific-catalogs/singlenode-kerberos-hdfs-impersonation/hive.properties (hunk 2 above)
Comment (arhimondr) [inline presto-product-tests/conf/presto/etc/environment-specific-catalogs/singlenode-kerberos-hdfs-impersonation/hive.properties]: Hive metastore impersonates HDFS operations to this user. So, it must have permissions to do everything withing `/user/hive/warehouse` directory. Which in our case belongs to `hive`

File: presto-product-tests/conf/presto/etc/environment-specific-catalogs/singlenode-kerberos-hdfs-impersonation/hive.properties (hunk 1 above)
Comment (cawallin) [inline presto-product-tests/conf/presto/etc/environment-specific-catalogs/singlenode-kerberos-hdfs-impersonation/hive.properties]: Ah ok, didn't know that was how Hive handled it. As a side note, why do we make the user specify it then?

**Comment by amrutagokhale** [comment]
Can you please substitute my existing commit in here with this https://github.com/Teradata/presto/pull/458/commits/d0e6a33ea963984a1e7c743a917f52541a369c59?

**Comment by arhimondr** [comment]
Sure, will do

**Comment by ywskycn** [comment]
As mentioned in #7107 , we also need to change the "getHostName()" to "getCanonicalHostName()" in KerberosAuthentication.java.

**Pull request merged**